source/base/data_out_base.cc

```diff
@@ -2371,6 +2432,26 @@ namespace DataOutBase
 
     for (const auto &patch : patches)
       {
+        // special treatment of simplices since they are not subdivided, s.t,
+        // no new nodes have to be created, but the precomputed ones can be
+        // used
+        if (patch.reference_cell_type != ReferenceCell::get_hypercube(dim))
+          {
+            Point<spacedim> node;
+
+            for (unsigned int point_no = 0; point_no < patch.data.n_cols();
+                 ++point_no)
+              {
+                for (unsigned int d = 0; d < spacedim; ++d)
+                  node[d] =
+                    patch.data(patch.data.size(0) - spacedim + d, point_no);
+
+                out.write_point(count++, node);
+              }
```

Comment:
Personally, I would like to keep on having the specialization since it also works (for mixed meshes) and older versions of Paraview.

Reply:
You mean the specialization for the quadratic version of those or the plain linear one? It is clear that we cannot use the tensor product variant below for simplices. But then my comment simply boils down to the question whether we can restructure the code and properly indent by `if` / `else` rather than `if ... continue`.

Reply:
Done!